lambdas/roleManager.js

```diff
@@ -5,70 +5,78 @@ const utils = require('./utils');
 module.exports.handler = async (event, context, callback) => {
     console.log('Received event: ', JSON.stringify(event, null, 2));
 
-    if (!event || !event.Records[0] || !event.Records[0].body) {
+    if (!event) {
         callback(new Error('No event specified'));
         return;
     }
 
-    const req = JSON.parse(event.Records[0].body);
-    // Validate RequestType
-    if (!req.RequestType || (req.RequestType !== 'Create' && req.RequestType !== 'Delete')) {
-        callback(new Error('Invalid RequestType. Valid values: "Create" or "Delete"'));
-        return;
-    }
-
-    // Check if all required attributes are set
-    if (!req.IntegrationName || !req.AccountIds || !req.UptAccountId || !req.ExternalId || !req.RequestType) {
-        callback(new Error(`Request:${JSON.stringify(req, null, 2)} is not valid. Required fields are missing`));
-        return;
-    }
+    for (let i = 0; i < event.Records.length; i++) {
 
-    // Currently we only support one account at a time
-    if (req.AccountIds.length !== 1) {
-        callback(new Error('AccountIds must be an array of strings with length of 1'));
-        return;
-    }
-
-    const stsCreds = await utils.stsCreds(req.AccountIds[0]);
-    const iamClient = await utils.iamClient(stsCreds, 'aws-global');
-    const sqsClient = await utils.sqsClient(event.Records[0].awsRegion);
+        if (!event.Records[i] || !event.Records[i].body) {
+            callback(new Error('No event specified'));
+            return;
```

Comment:
be careful with returns.
returning on Error is fine here after the callback.
Please review other places